### PR TITLE
Feat(evm-bridge): add whitelist to pay for evm transactions

### LINKED ISSUE #385: Implement whitelist functionality in bridge for gass-less transactions.
Some of dApps want to host own bridge and pay fee for their customers.

In order to make it pay off, dApp creators should implement fee logic in their contracts, which is out of scope of velas-chain project.

But also, bridge should allow setting token list that it will process, and pay velas fee for it.

Proposed whitelist format for bridge is an array of pairs (contract_address, input_data_filter)

Where input_data_filter, at first milestone can filter only by starting bytes.

`tx.input.start_with(input_data_filter)`

### PR DESCRIPTION
**Pull request by gBaGu**
Added `--whitelist-path` arg to bridge. It expects a path to json file with array of filters

Only "InputStartsWith" is supported so far.
Whitelist config example:
```
[
	{ "InputStartsWith": [ "0xC56c3022a16FC3f7eA7538034266D6aDf290B095", "0xe2a2d66a" ] }
]
```
This means that all transactions to 0xC56c3022a16FC3f7eA7538034266D6aDf290B095 that will have it's input starting with 0xe2a2d66a will be payed by a bridge.

### CODE REVIEW

File: evm-utils/evm-bridge/src/tx_filter.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+use evm_state::{Address, Transaction, TransactionAction};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub enum TxFilter {
+    InputStartsWith(Address, Bytes),
+}
+
+impl TxFilter {
+    pub fn filter(&self, tx: &Transaction) -> bool {
```
Comment (vldm) [inline evm-utils/evm-bridge/src/tx_filter.rs]: Usually "filter" name is used for reverse logic - when you need to remove something from collection that is not match.

I propose "match" or better "is_match" name for this method.

File: evm-utils/evm-bridge/src/tx_filter.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+use evm_rpc::Bytes;
+use evm_state::{Address, Transaction, TransactionAction};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub enum TxFilter {
+    InputStartsWith(Address, Bytes),
```
Comment (vldm) [inline evm-utils/evm-bridge/src/tx_filter.rs]: Maybe add some docs about this enum, or/and use Struct variant to add some mining to fields.
```suggestion
    InputStartsWith {
        contract: Address,
        input_prefix: Bytes
     },
```

File: evm-utils/evm-bridge/src/tx_filter.rs (hunk 1 above)
Comment (gBaGu) [reply evm-utils/evm-bridge/src/tx_filter.rs]: Renamed to 'is_match'

File: evm-utils/evm-bridge/src/tx_filter.rs (hunk 2 above)
Comment (gBaGu) [reply evm-utils/evm-bridge/src/tx_filter.rs]: Done

**Review by vldm — APPROVED** [review]

**Pull request merged**